packages/immutadot/src/nav/arrayNav.js

```diff
@@ -0,0 +1,15 @@
+import { isNil, length } from 'util/lang'
+import { BaseNav } from './baseNav'
+
+export class ArrayNav extends BaseNav {
+  get length() {
+    if (this._length === undefined) this._length = length(this.value)
+    return this._length
+  }
+
+  copy() {
+    const { value } = this
+    if (isNil(value)) return []
+    return Array.isArray(value) ? [...value] : { ...value }
```

Comment:
Because users might slice or array index notation on an object, this is to avoid replacing the object by an array.

This conforms to normal JavaScript behavior:
```js
const obj = {}
obj[0] = 'val'
console.log(obj) // Outputs: {0: "val"}
```